content/news/posts/2024/meet-prestashop9-api.md

```diff
@@ -0,0 +1,386 @@
+
+```json
+{
+    "customerGroupId": 4,
+    "localizedNames": {
+        "1": "Nowa grupa",
+        "2": "New group"
+    },
+    "reductionPercent": 0.0,
+    "displayPriceTaxExcluded": false,
+    "showPrice": true,
+    "shopIds": [
+        1
+    ]
+}
+```
+
+and you will see the new user group in the PrestaShop back office.
+
+![New group added via the API available in the back office](/assets/images/2024/05/api/bo_groups.jpeg)
+
+If you don't provide all the necessary information, you will receive an error message. For example, if you try to add a new user group without providing the `localizedNames` parameter, you will receive the following response:
+
+![Error when adding a new resource without providing required information](/assets/images/2024/05/api/postman_add_group_error.png)
+
+## Deleting a single resource
+
+To delete a resource, you need to send a `DELETE` request to the `/admin-api/customers/group/{customerGroupId}` endpoint, where `{customerGroupId}` is the ID of the user group you want to delete.
+
+```bash
+curl --location --request DELETE 'http://yourdomain.test/admin-api/customers/group/4' \
+--header 'Authorization: Bearer YOUR_ACCESS_TOKEN'
+```
+
+## Importing API collection to your API client
+
+Thanks to the automatically generated documentation, you can easily import the API collection into Postman or another API client of your choice. You can find the collection in the `http://yourdomain.test/admin-api/docs.json` URL. The collection is available in the OpenAPI format, compatible with Postman and other API clients.
+
+## Extending API
+
+The new API in PrestaShop 9 is designed to be extensible. You can add new endpoints, scopes, and other features to the API. We plan to release extensive developer documentation for the API soon. In the meantime, you can check the [example module](#) and the [ps_apiresources](#).
+
+{{% notice %}}
+The new API in PrestaShop 9 is still under development. We encourage you to test it and provide feedback to help us improve it further.
+{{% /notice %}}
+
+## Get involved
+
+The new API in PrestaShop 9 is a significant step forward for the platform. It will provide developers with a modern, secure, and extensible API that will allow them to create new integrations and services. We encourage you to test the new API and provide feedback to help us improve it further.
+
+If you are motivated to contribute to developing the new API, we encourage you to join the PrestaShop community. More information about contributing to PrestaShop is on the [Contribute to PrestaShop](https://devdocs.prestashop-project.org/9/contribute/) page.
+
+### Let's keep in touch on Slack
+
+If you have any questions or feedback, come [talk with us on Slack](https://www.prestashop-project.org/slack/). We are always happy to help you with any issues you may have. You can join the `#dev-core` channel to discuss the new API in PrestaShop 9.
```

Comment:
Should we have information about the fact that the Authorization server is also modular? Like with our keycloak example module? Or is it too advanced for the purpose of this article?

Reply:
We can mention that, good idea. I'll also mention that more details on how to use it will be available in the developer documentation.